src/main/java/com/guciowons/shoppingbasket/Basket/BasketSummarizer.java

```diff
@@ -0,0 +1,35 @@
+package com.guciowons.shoppingbasket.Basket;
+
+import com.guciowons.shoppingbasket.Product.ProductDao;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+@Component
+public class BasketSummarizer {
+    public BasketSummarized summarizeBasket(HashMap<Integer, Integer> basketProducts, ProductDao productDao){
```

Comment:
Tutaj przekazanie `ProductDao` jest trochę na wyrost. Po to masz springa, żeby przerzucanie beanów było prostrze, więc:
- albo wstrzykujesz `ProductDao` tutaj tak jak wszędzie indziej (w końcu to też zależność między beanami)
- albo do metody przekazujesz tylko listę produktów, która już była pobrana z dao -- ta opcja chyba lepsza bo masz jedno powiązanie beanow mniej